Gui/opensim/view/src/org/opensim/view/experimentaldata/ExperimentalMarker.java

```diff
@@ -104,9 +107,11 @@ public JSONObject createDecorationJson(ArrayList<UUID> comp_uuids, MotionDisplay
         StateVector dataAtStartTime = motionDisplayer.getSimmMotionData().getStateVector(0);
         ArrayDouble interpolatedStates = dataAtStartTime.getData();
         int idx = getStartIndexInFileNotIncludingTime();
+        AnnotatedMotion mot = (AnnotatedMotion)motionDisplayer.getSimmMotionData();
```

Comment:
This is just a cast. Creation is indicated by calling 'new'

Reply:
Then how is `Storage` a valid  `AnnotatedMotion`? How are the additional members of the `AnnotatedMotion` valid? Are they initialized to default values? How do the annotations have meaning?

Reply:
Storage is not a valid AnnotatedMotion, it's the otherway around. Except we know in the case we diplay experimental markers that the data was annotated and thus we can downcast.

Reply:
@aseth1 that's built into the language, the objects keep their true type rather than get "sliced" since they are just references.  If an object is not the right subclass it will throw exception.

Reply:
Still confused. So `getSimmMotionData()` does in fact return an `AnnotatedMotion`? Then that makes sense, but how would one know that?

Reply:
How/why is `motionDisplayer.getSimmMotionData()` guaranteed to return an `AnnotatedMotion`.  That is certainly not obvious.

Reply:
It's not guaranteed, in other cases (e.g. showing normal motions it isn't). If it did I'd have built it into the signature. If it wasn't in this case an exception is thrown.